routes/bridges/gassistant/index.js

```diff
@@ -0,0 +1,235 @@
+        if (title.length <= 25)
+            suggestions.push(title.substring(0, 25));
+        else
+            console.log(`${title} exceeds max length of 25 characters`);
+        if (suggestions.length)
+            this._buffer.push(new Suggestions(suggestions));
+    }
+
+    sendLink(title, url) {
+        if (url === '/user/register') {
+            this._requestSignin = true;
+        } else {
+            this._buffer.push(new Button({
+                title: title,
+                url: url,
+            }));
+        }
+    }
+
+    sendResult(message, icon) {
+        if (typeof this._buffer[this._buffer.length - 1] === 'string')
+            // If there is already a text reply immediately before, we merge text replies
+            // because Google Assistant limits at most 2 chat bubbles per turn
+            this._buffer[this._buffer.length - 1] += '\n' + message.toLocaleString(this._locale);
+        else
+            this._buffer.push(message.toLocaleString(this._locale));
+    }
+
+    sendAskSpecial(what) {
+        // TODO
+    }
+}
+GoogleAssistantDelegate.prototype.$rpcMethods = ['send', 'sendPicture', 'sendChoice', 'sendLink', 'sendButton', 'sendAskSpecial', 'sendRDL', 'sendResult'];
+
+function authenticate(req, res, next) {
+    console.log(req.body.user);
+    if (req.body.user.accessToken) {
+        req.headers.authorization = 'Bearer ' + req.body.user.accessToken;
+        passport.authenticate('bearer', { session: false })(req, res, next);
+    } else {
+        next();
+    }
+}
+
+router.use(authenticate);
+router.use(userUtils.requireScope('user-exec-command'));
+
+const app = actionssdk();
+
+async function retrieveUser(accessToken) {
+    let anonymous, user;
+    if (accessToken) {
+        const decoded = await util.promisify(jwt.verify)(accessToken, secret.getJWTSigningKey(), {
+            algorithms: ['HS256'],
+            audience: 'oauth2',
+            clockTolerance: 30,
+        });
+        user = await db.withClient(async (dbClient) => {
+            const rows = await model.getByCloudId(dbClient, decoded.sub);
+            if (rows.length < 1) {
+                anonymous = true;
+                return await userUtils.getAnonymousUser();
+            }
+            await model.recordLogin(dbClient, rows[0].id);
+            anonymous = false;
+            return rows[0];
+        });
+    } else {
+        anonymous = true;
+        user = await userUtils.getAnonymousUser();
+    }
+    return [anonymous, user];
+}
+
+// Welcome response when user first initiates conversation
+app.intent('actions.intent.MAIN', async (conv) => {
+
+    const [anonymous, user] = await retrieveUser(conv.body.user.accessToken);
+
+    const locale = conv.body.user.locale;
+    const conversationId = conv.body.conversation.conversationId;
+    const assistantUser = { name: conv.user.name.display || 'User', isOwner: true };
+    const delegate = new GoogleAssistantDelegate(locale);
+
+    return Q.try(() => {
```

Comment:
Q.try is part of really old library which we should not use. You're already inside an async function, you don't need this.

Reply:
Fixed